iterator.md

```diff
@@ -0,0 +1,189 @@
+# Iterator-Pattern
+
+## Definition
+
+Das Iterator-Pattern wird verwendet, um über eine Liste zu traversieren ohne hier entsprechend die Struktur oder das Muster zu
+veröffentlichen. Heißt, eine Liste nach und nach Abzugehen. Der Iterator ist in diesem Fall der `Cursor`, welche durch die Kollektion
+durchgeht.
+
+## UML
+
+![alt text](iterator_uml.png)
+
+## Anwendung
+
+Um einen Iterator Anwenden zu können, muss zuvor ein Iterator Interface angelegt werden, wo die entsprechenden Methoden (next, 
+first,isDone und CurrentItem) erzeugt werden. Anschließend wird ein ConcreteIterator als Klasse definiert, damit der Iterator weiß, an 
+welcher Stelle dieser sich gerade befindet.
+
+Beispielcode für die Erstellung und Erzeugung eines Iterator-Pattern:
+
+### Erzeugen des Iterator Interfaces
+
+```csharp
+interface IIterator
+{
+    string FirstItem { get;}
+    string NextItem{ get;}
+    string CurrentItem{ get;}
+    bool IsDone { get;}
+}
+```
+
+### Erzeugen des Interfaces für die Aggregator Kollektion
+
+```csharp
+interface IAggregate
+{
+    IIterator GetIterator();
+    string this[int itemIndex]{set;get;}
+    int Count{get;}
+}
+```
+
+#### Erzeugen der Konkreten Klasse für die Aggreator Kollektion
+
+```csharp
+class MyAggregate : IAggregate
+{
+    List<string> values_ = null;
```

Comment:
wieso der _ am ende der variable?

Reply:
der kann theoretisch auch weg.. ich hatte das Beispiel von einer Seite genommen wo es recht gut erklärt war.. zur not wie gesagt lösche ich den überall